src/components/layout/CdrLayout.vue

```diff
@@ -0,0 +1,87 @@
+<script setup lang="ts">
+import { useCssModule, computed } from 'vue';
+import mapClasses from '../../utils/mapClasses';
+import { getStructureStyles } from '../../utils/layout';
+import type { Layout, NameValuePair } from '../../types/interfaces';
+import type { Structure } from '../../types/other';
+import { modifyClassName } from '../../utils/buildClass';
+
+/** Foundational container for creating structured layouts */
+
+defineOptions({ name: 'CdrLayout' });
+
+const props = withDefaults(defineProps<Layout>(), {
+  gap: 'zero',
+  rowGap: 'zero',
+  columnGap: 'zero',
+  columns: undefined,
+  rows: undefined,
+  as: 'div',
+  queryType: 'container',
+  flow: undefined,
+  flowValue: 'auto',
+});
+
+const style = useCssModule();
+
+const rootProps = computed(() => {
+  const baseClass = 'cdr-layout';
+  const classes = [baseClass];
+  const inlineStyles: NameValuePair = {};
+
+  // Add gap for entire grid
+  if (props.gap !== 'zero') {
+    classes.push(modifyClassName(baseClass, `gap-${props.gap}`));
+  }
+
+  // Add gap for columns
+  if (props.columnGap !== 'zero') {
+    classes.push(modifyClassName(baseClass, `column-gap-${props.columnGap}`));
+  }
+
+  // Add gap for rows
+  if (props.rowGap !== 'zero') {
+    classes.push(modifyClassName(baseClass, `row-gap-${props.rowGap}`));
+  }
+
+  // Add grid-auto-flow and supporting value
+  if (props.flow) {
+    classes.push(modifyClassName(baseClass, `flow-${props.flow}`));
+    inlineStyles['--cdr-layout-flow-value'] = props.flowValue;
+  }
+
+  // Add grid templates for columns and rows
+  (['rows', 'columns'] as Structure[]).forEach((structure) => {
+    if (!props[structure]) {
+      return;
+    }
+
```

Comment:
Chatted about in meeting: this is something we will work in (if appropriate) when we start building out filmstrip. My initial suggestion is creating a Viewport component that handles the scrolling/controls, rather than include in Layout.